### PR TITLE
Fix more violations of strictNullChecks

### PR DESCRIPTION
**Pull request by AE-MS**
## Description

- Fix some places where `undefined` / `null` was being handled but it just wasn't part of the type definition (by adding it to the type definition)
- Fix the code which assesses the `runtime` object to detect `undefined` cases and throw in those.

### Validation performed:

Run unit tests and e2e tests as part of this PR

### Unit Tests added:

Yes, added a unit test to cover the one new behavior: if "null" is passed as the runtime config stored in `clientSupportedSDKVersion` (the flipped order scenario). This shouldn't ever happen in real life but the type system allows it, and now we protect against it.

### End-to-end tests added:

No, there are no e2e tests that would exercise functionality not already covered by unit tests.

### Change file added:

Yes

### Related PRs:

https://github.com/OfficeDev/microsoft-teams-library-js/pull/1356
https://github.com/OfficeDev/microsoft-teams-library-js/pull/1352

### Next/remaining steps:

Fix more violations

### CODE REVIEW

File: packages/teams-js/src/public/app.ts [hunk 1]
```diff
@@ -589,9 +588,14 @@ export namespace app {
...
                     GlobalVars.clientSupportedSDKVersion = runtimeConfig;
                   }
-                  /* eslint-disable-next-line strict-null-checks/all */ /* Fix tracked by 5730662 */
-                  const givenRuntimeConfig: IRuntime = JSON.parse(clientSupportedSDKVersion);
-                  clientSupportedSDKVersion && applyRuntimeConfig(givenRuntimeConfig);
+                  const givenRuntimeConfig: IRuntime | null = JSON.parse(clientSupportedSDKVersion);
+                  initializeHelperLogger('givenRuntimeConfig parsed to %o', givenRuntimeConfig ?? 'null');
+
+                  if (!givenRuntimeConfig) {
+                    throw new Error('givenRuntimeConfig successfully parsed to null');
```
Comment (erinha) [inline packages/teams-js/src/public/app.ts:595]: This seems like a slightly weird error message containing the word "successfully". Suggest removing the word "successfully".

**Review by erinha — CHANGES_REQUESTED** [review]
Comment about error message. Let me know if you disagree!

File: packages/teams-js/src/public/app.ts (hunk 1 above)
Comment (AE-MS) [reply packages/teams-js/src/public/app.ts:595]: Interesting; I actually intentionally used that word to indicate that the parsing was successful and it was not the result of a parsing error. 🤔

Comment (erinha) [reply packages/teams-js/src/public/app.ts:595]: I wondered about that perspective and since I don't have a better suggestion, I guess it's fine.

**Review by erinha — DISMISSED** [review]

File: packages/teams-js/src/public/app.ts (hunk 1 above)
Comment (AE-MS) [reply packages/teams-js/src/public/app.ts]: Maybe this new string is clearer?

**Review by erinha — APPROVED** [review]
Thank you, this error message is definitely an improvement IMO!

**Review by TrevorJoelHarris — APPROVED** [review]

**Pull request merged**